tests/unit/type/hash-field-expire.tcl

```diff
@@ -0,0 +1,586 @@
+######## HEXPIRE family commands
+# Field does not exists
+set E_NO_FIELD     -2
+# Specified NX | XX | GT | LT condition not met
+set E_FAIL         0
+# expiration time set/updated
+set E_OK           1
+# Field deleted because the specified expiration time is in the past
+set E_DELETED      2
+
+######## HTTL family commands
+set T_NO_FIELD    -2
+set T_NO_EXPIRY   -1
+
+######## HPERIST
+set P_NO_FIELD    -2
+set P_NO_EXPIRY   -1
+set P_OK           1
+
+############################### AUX FUNCS ######################################
+
+proc create_hash {key entries} {
+    r del $key
+    foreach entry $entries {
+        r hset $key [lindex $entry 0] [lindex $entry 1]
+    }
+}
+
+proc get_keys {l} {
+    set res {}
+    foreach entry $l {
+        set key [lindex $entry 0]
+        lappend res $key
+    }
+    return $res
+}
+
+proc cmp_hrandfield_result {hash_name expected_result} {
+    # Accumulate hrandfield results
+    unset -nocomplain myhash
+    array set myhash {}
+    for {set i 0} {$i < 100} {incr i} {
+        set key [r hrandfield $hash_name]
+        set myhash($key) 1
+    }
+     set res [lsort [array names myhash]]
+     if {$res eq $expected_result} {
+        return 1
+     } else {
+        return $res
+     }
+}
+
+proc hrandfieldTest {activeExpireConfig} {
+    r debug set-active-expire $activeExpireConfig
+    r del myhash
+    set contents {{field1 1} {field2 2} }
+    create_hash myhash $contents
+
+    set factorValgrind [expr {$::valgrind ? 2 : 1}]
+
+    # Set expiration time for field1 and field2 such that field1 expires first
+    r hpexpire myhash 1 NX 1 field1
+    r hpexpire myhash 100 NX 1 field2
+
+    # On call hrandfield command lazy expire deletes field1 first
+    wait_for_condition 8 10 {
+        [cmp_hrandfield_result myhash "field2"] == 1
+    } else {
+        fail "Expected field2 to be returned by HRANDFIELD.
```

Comment:
```suggestion
        fail "Expected field2 to be returned by HRANDFIELD."
```

Reply:
amazing how TCL parser simply ignores it and carry on